runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/TestDataflowRunner.java

```diff
@@ -134,8 +131,6 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
       throw new RuntimeException(errorMessage(job, messageHandler));
     }
 
-    // If there is no reason to immediately fail, run the success matcher.
-    assertThat(job, testPipelineOptions.getOnSuccessMatcher());
```

Comment:
@iemejia given that we need to deprecate `TestPipelineOptions` and not remove it, is it ok to remove the few places where the matchers are consumed (in `TestDataflowRunner` and `TestSparkRunner`), or is there a concern that that will break some user tests?